test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -420,43 +420,58 @@ TEST_VM(os, realpath) {
   const char* returnedBuffer = os::realpath(nosuchpath, buffer, sizeof(nosuchpath) - 2);
   /* Returns ENOENT on Linux, ENAMETOOLONG on Windows */
   EXPECT_TRUE(returnedBuffer == nullptr);
-  EXPECT_TRUE(errno == ENAMETOOLONG || errno == ENOENT);
+#if defined(_WINDOWS)
+  EXPECT_TRUE(errno == ENAMETOOLONG);
+#else
+  EXPECT_TRUE(errno == ENOENT);
+#endif
 
   errno = 0;
   returnedBuffer = os::realpath(nosuchpath, buffer, sizeof(nosuchpath) + 3);
   /* Returns ENOENT on Linux, 0 on Windows */
-  EXPECT_TRUE(errno == 0 || errno == ENOENT);
-  EXPECT_TRUE(returnedBuffer == nullptr || returnedBuffer == buffer);
+#if defined(_WINDOWS)
+  EXPECT_TRUE(returnedBuffer == buffer);
+  EXPECT_TRUE(errno == 0);
+#else
+  EXPECT_TRUE(returnedBuffer == nullptr);
+  EXPECT_TRUE(errno == ENOENT);
+#endif
 
   errno = 0;
   returnedBuffer = os::realpath(tmppath, buffer, MAX_PATH);
-  EXPECT_TRUE(returnedBuffer == buffer);
+  EXPECT_TRUE(returnedBuffer != nullptr);
```

Comment:
Why the change?

Reply:
An earlier misundertanding had me thinking the buffer allocated by realpath could be returned.  I've reverted this change.